src/Sentry.Maui.CommunityToolkit.Mvvm/CtMvvmMauiElementEventBinder.cs

```diff
@@ -0,0 +1,157 @@
+using System.Windows.Input;
+using CommunityToolkit.Mvvm.Input;
+
+namespace Sentry.Maui.CommunityToolkitMvvm;
+
+/// <summary>
+/// Scans all elements for known commands that are implement
+/// </summary>
+public class CtMvvmMauiElementEventBinder(IHub hub) : IMauiElementEventBinder
+{
+    /// <summary>
+    /// Binds to the element
+    /// </summary>
+    /// <param name="element"></param>
+    /// <param name="addBreadcrumb"></param>
+    public void Bind(VisualElement element, Action<BreadcrumbEvent> addBreadcrumb) => Iterate(element, true);
+
+    /// <summary>
+    /// Unbinds from the element
+    /// </summary>
+    /// <param name="element"></param>
+    /// <exception cref="NotImplementedException"></exception>
+    public void UnBind(VisualElement element) => Iterate(element, false);
+
+
+    private void Iterate(VisualElement element, bool bind)
+    {
+        switch (element)
+        {
+            case Button button:
+                TryBindTo(button.Command, bind);
+                break;
+
+            case ImageButton imageButton:
+                TryBindTo(imageButton.Command, bind);
+                break;
+
+            case CarouselView carousel:
+                TryBindTo(carousel.CurrentItemChangedCommand, bind);
+                TryBindTo(carousel.RemainingItemsThresholdReachedCommand, bind);
+                TryBindTo(carousel.PositionChangedCommand, bind);
+                break;
+
+            case CollectionView collectionView:
+                TryBindTo(collectionView.RemainingItemsThresholdReachedCommand, bind);
+                TryBindTo(collectionView.SelectionChangedCommand, bind);
+                break;
+
+            case Entry entry:
+                TryBindTo(entry.ReturnCommand, bind);
+                break;
+
+            case RefreshView refresh:
+                TryBindTo(refresh.Command, bind);
+                break;
+
+            case SearchBar searchBar:
+                TryBindTo(searchBar.SearchCommand, bind);
+                break;
+
+            default:
+                TryGestureBinding(element, bind);
+                break;
+        }
+    }
+
+    private void TryGestureBinding(VisualElement element, bool bind)
+    {
+        if (element is IGestureRecognizers gestureRecognizers)
+        {
+            foreach (var gestureRecognizer in gestureRecognizers.GestureRecognizers)
+            {
+                TryBindTo(gestureRecognizer, bind);
+            }
+        }
+    }
+
+    private void TryBindTo(IGestureRecognizer recognizer, bool bind)
+    {
+        switch (recognizer)
+        {
+            case TapGestureRecognizer tap:
+                TryBindTo(tap.Command, bind);
+                break;
+
+            case SwipeGestureRecognizer swipe:
+                TryBindTo(swipe.Command, bind);
+                break;
+
+            // no commands
+            //case PinchGestureRecognizer pinch
+            //case PanGestureRecognizer pan
```

Comment:
Is the commented code included as an explanation or a todo or could we remove it?

Reply:
It was left as somewhat of a note for implementors in the future.  There are no commands on those, so I wanted to make it known why they were missing.  I've updated the note and moved it to the bottom of the switch